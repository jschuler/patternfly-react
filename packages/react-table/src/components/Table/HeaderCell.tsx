import * as React from 'react';
import { BaseHeaderCell } from '../BaseTable/BaseHeaderCell';

export interface HeaderCellProps {
  'data-label'?: string;
  className?: string;
  component?: React.ReactNode;
  isVisible?: boolean;
  scope?: string;
  textCenter?: boolean;
  dataLabel?: string;
  tooltip?: string;
  onMouseEnter?: (event: any) => void;
  children: React.ReactNode;
}

export const HeaderCell: React.FunctionComponent<HeaderCellProps> = ({
  className = '',
  component = 'th',
  scope = '',
  textCenter = false,
  tooltip = '',
  onMouseEnter = () => {},
  children,
  /* eslint-disable @typescript-eslint/no-unused-vars */
  isVisible,
  dataLabel = '',
  /* eslint-enable @typescript-eslint/no-unused-vars */
  ...props
}: HeaderCellProps) => (
  <BaseHeaderCell
    {...props}
    scope={scope}
    tooltip={tooltip}
    onMouseEnter={onMouseEnter}
    textCenter={textCenter}
    component={component}
    className={className}
  >
    {children}
  </BaseHeaderCell>
);
HeaderCell.displayName = 'HeaderCell';
