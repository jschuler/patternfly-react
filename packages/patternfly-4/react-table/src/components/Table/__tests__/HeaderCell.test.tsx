// Auto-generated do not edit

import * as React from 'react';
import { shallow } from 'enzyme';
import { HeaderCell } from '../HeaderCell';

describe('HeaderCell test', () => {
  it('HeaderCell should match snapshot', () => {
    const view = shallow(
      <HeaderCell
        data-label={undefined}
				className={"''"}
				component={<>ReactNode</>}
				isVisible={undefined}
				scope={"''"}
				textCenter={false}
				dataLabel={"''"}
      />);
    expect(view).toMatchSnapshot();
  });
});
