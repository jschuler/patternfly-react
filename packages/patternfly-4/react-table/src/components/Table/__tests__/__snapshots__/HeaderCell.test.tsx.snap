// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderCell test HeaderCell should match snapshot 1`] = `
<[object Object]
  className="''"
  scope="''"
/>
`;
